Make Create Actions open as dialogs (part 2)

madsrasmussen opened the pull request:
Fixes: https://github.com/umbraco/Umbraco-CMS/issues/20516

Updates script and Blueprint dialogs that were missed in this PR: https://github.com/umbraco/Umbraco-CMS/pull/20489

This PR fixes the script create dialog:
<img width="1275" height="869" alt="Screenshot 2025-10-16 at 12 02 18" src="https://github.com/user-attachments/assets/7817ab8e-8f8e-4b9f-b2b4-d79b2b2bf403" />

And updates the Blueprint dialog to match the other create option dialogs. Instead of displaying the Document Type Tree directly in the dialog, it is now a two-step process: first select create blueprint, then choose the document type.

<img width="1271" height="868" alt="Screenshot 2025-10-16 at 12 02 30" src="https://github.com/user-attachments/assets/b07646bc-f506-4fda-b683-1e99f9f550ed" />

<img width="1273" height="865" alt="Screenshot 2025-10-16 at 12 02 40" src="https://github.com/user-attachments/assets/3c6e5886-eb88-43fd-a47c-1f3c43d983b7" />

AndyButland reviewed (commented):
Looks good to me testing out locally.

I noticed the Stylesheets create dialog, whilst being centred, is a little different to the scripts one.  The "New Folder" item doesn't have the "..." and it does have a description.  Think it would make sense to align this with what you've done for Scripts and Blueprints.

Also, in the linked issue is raised the point about when creating a member, even if there's only one member type, you still see the dialog.  Not sure how you want to handle that - in this PR, ask for a separate issue to be created, or just leave it as is?  But just raising so we don't miss that part of the issue.

madsrasmussen commented:
@AndyButland I have pushed some additional updates to this one to align the dialogs more. I have removed the word "New" from all the options I could find. We don't use "New" when creating documents, media, etc., so I think this aligns better. Additionally, I have fixed the implementation of the ellipsis. We already supported the "additionalOptions" setting in the manifest but never implemented it in the dialog. If we need further alignment, I think we should invest time in correctly registering all options as an "entityCreateOptionAction" so they all go through the same dialog/UI and we can remove a bunch of custom dialogs.

The PR doesn't target the single option dialog. I think it could make sense that the Management Api returns the "allowedChildren" information from the Content Type on a Document Tree Model. This would give us some extra capabilities on the frontend:
- Check if we have any options at all (remove create option, if you can't create anything).
- Only show the dialog if there is more than one option
- When moving/copying, we have quick access to where we can move or copy to.

@andr317c Will you help fix merge conflicts in this one? 🙏

AndyButland reviewed (approved):
All looks good to me from the perspective of the dialogs now.

I've taken a note about the suggestion on the content responses from the trees including the allowed children - we can look at the feasibility of that in a separate PR.  My only concern is that it's another database look up, which is probably why we currently get this information only having clicked on "Create".

Just one tiny thing, mentioning only if easy to do something about, is that there's a small but perceptible flicker on the document and media create dialogs where the briefly appear in the left and then move to the centre.  I don't see it on the dialogs under "Settings".  Might just be it's too quick to see.  No reason that should hold up this PR though.

madsrasmussen commented:
@AndyButland I have noticed the dialog flicker too. It is a problem with all dialogs, so I would like to keep it out of this PR.
